### PR TITLE
Avoid Split(new char[]) inside loops

### PR DESCRIPTION
**Pull request by alaatm**

### CODE REVIEW

**Comment by GrabYourPitchforks** [comment]
Are you sure these changes are necessary? Some projects (like DirectoryServices) cross-compile against .NET Full Framework, so they won't be able to use this API. Other projects like the STEW Unicode helpers are tools that are run only once on-demand every release.

**Comment by alaatm** [comment]
@GrabYourPitchforks Thanks for the insights, make sense. I am closing this.

File: src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapDirectoryIdentifier.cs [hunk 1]
```diff
@@ -35,7 +35,7 @@ public LdapDirectoryIdentifier(string[] servers, bool fullyQualifiedDnsHostName,
                     if (servers[i] != null)
                     {
                         string trimmedName = servers[i].Trim();
-                        string[] result = trimmedName.Split(new char[] { ' ' });
+                        string[] result = trimmedName.Split(' ');
```
Comment (stephentoub) [inline src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapDirectoryIdentifier.cs]: I think this is a reasonable change. It'll bind to different overloads based on the target framework and at least save on the array allocation when built for core.

File: src/libraries/System.DirectoryServices/src/System/DirectoryServices/DirectoryEntry.cs [hunk 2]
```diff
@@ -1003,7 +1003,7 @@ public void RefreshCache(string[] propertyNames)
                         _propertyCollection.valueTable.Remove(name);
 
                         // also need to consider the range retrieval case
-                        string[] results = name.Split(new char[] { ';' });
+                        string[] results = name.Split(';');
```
Comment (stephentoub) [inline src/libraries/System.DirectoryServices/src/System/DirectoryServices/DirectoryEntry.cs]: Ditto

**Review by stephentoub — COMMENTED** [review]
The two changes I highlighted seem reasonable. I agree with Levi on the rest that they're not worthwhile. Thanks.

**Pull request closed without merging**